### DISCUSSION TITLE
How to show report after press button

### DISCUSSION THREAD

**Discussion by haajit** [serenity-is/Serenity#6351] [Help] [answered]
Hi
I am beginner in serenity
and by use of this wiki I could show a report by use URL: http://localhost:5555/Reports
but don't know how to show report after press this button

![image](https://user-images.githubusercontent.com/9511137/161474151-2ce33e87-3f7b-4e6e-9866-00c16aa09d11.png)

**Comment by minhhungit** [comment]
@haajit  Inside your button click event, call it likes this:

```javascript
let opt: Common.ReportExecuteOptions = {
      reportKey: 'Put_Your_Report_Key_Here',
      extension: 'pdf',
      getParams: () => ({
          FirstParam: 'abc',
          SecondParam: 'xyz',
      })
};

Common.ReportHelper.execute(opt);
```

**Reply by haajit** [reply]
Hi [minhhungit](https://github.com/minhhungit)
thanks for your answer
I owe you an apology
I forgot to say that I want to use telerik for reporting by wiki : https://github.com/serenity-is/Serenity/wiki/Reporting-Integrating-Telerik-Reporting

**Reply by minhhungit** [reply]
Then create your telerik report page and open the page in your button click event 

check this page: https://www.geeksforgeeks.org/how-to-open-url-in-new-tab-using-javascript/#:~:text=To%20open%20a%20new%20tab,or%20null%20if%20it%20failed.

for example

window.open(
Q.resolveUrl(`~/Default/YourPageHere?firstParam=${item.Id}&secondParam=${item.Something}`), "_blank");

**Reply by minhhungit** [reply]
or your case 
```window.open(Q.resolveUrl("~/Reports"), "_blank")```

hope it works

**Reply by haajit** [reply]
Thanks alot
It solved my problem

**Comment by haajit** [comment, marked as answer]
By:

            Q.postToUrl({
                url: '~/ReceptionPrint',
                params: {
                    COHSysCode: this.form.CohSysCode.value
                },
                target: "_blank"
            });
I was able to send a parameter and show the report.
